.github/workflows/workflow.yml

```diff
@@ -1,38 +1,53 @@
 name: CI
 
 on:
-  push:
+  pull_request:
     branches:
       - master
-  pull_request:
+  push:
     branches:
       - master
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - macos-latest
+          - ubuntu-latest
+        ocaml-version:
+          - 4.10.1
+
+    runs-on: ${{ matrix.os }}
+
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
 
-      - name: Cache
+      - name: Retrieve opam cache
         uses: actions/cache@v2
+        id: cache-opam
         with:
-          path: /home/runner/.opam
-          key: ez-api-opam-cache
+          path: ~/.opam
+          key: v1-${{ runner.os }}-opam-${{ matrix.ocaml-version }}-${{ hashFiles('ez-api.opam') }}
```

Comment:
> Caches are immutable so it is useful to start all your cache keys with a version prefix, for example `v1-...` This enables you to regenerate all of your caches by incrementing the version in this prefix. [Caching Dependencies: Managing Caches - CircleCI](https://circleci.com/docs/2.0/caching/#managing-caches)

Reply:
Ideally, we should use opam lock to create a more strict key, but I don't think there's a big problem with the current one.